### DISCUSSION TITLE
Coordinate transformation and stiffness difference

### DISCUSSION THREAD

**Discussion by vacupenda** [wenbinyugroup/.github#6] [VABS Related Discussions] [answered]
When I first ran the ". sg" file using VABS, there was a significant difference in the calculated bending stiffness compared to the coordinate "principal bending axes", and at this point, the ". k" file was mentioned as follows
"The principal bending axes rotated from the user coordinate system by    2.5713574208102119      degrees about the positive direction of x1 axis."
So I rotated the original coordinate system:<rotate>-2.5713574208102119</rotate>, but the final result still had a significant difference. At this point, ". k" has the following prompt: "The principal bending axes rotated from the user coordinate system by 1.684445858941341E-005 degrees about the positive direction of x1 axis." "
So what should I do to obtain the stiffness matrix under "principal bending axes"

![4](https://github.com/wenbinyugroup/.github/assets/135616110/d8495567-f162-4415-9bfb-20b261384850)

**Comment by unitasium** [comment]
In general, C33 and C44 components of the stiffness matrix should not be directly interpreted as EI22 and EI33. A proper way to calculate them is to use the compliance matrix: EI22=1/S33 and EI33=1/S44.

After the rotation, the stiffness matrix is actually aligned with the principal bending axes. If you want C33=EI22 and C44=EI33, then you need to decouple bending from all other dofs. In this case, you may also need to move the origin of the cross-section to the tension center to decouple EA and EI22/EI33.

**Reply by vacupenda** [reply]
Thank you for your reply, I have moved  the origin of the cross-section to the tension center, But in the end, the diagonal elements of the "Stiffness Matrix" are "C33 ≠ EI22 and C44 ≠ EI33". In the "Compliance Matrix", GJ=1/S22, EI22=1/S33, E133=1/S44 but EA ≠ 1/S11, is it because there is an issue with my coordinate conversion?
![5](https://github.com/wenbinyugroup/.github/assets/135616110/d6a40249-d659-4ddb-9432-f24a72e2360b)

**Reply by unitasium** [reply, marked as answer]
You can see that after translation, the two coordinates of the tension center are larger, which means that the center is further away from the origin. There are two things one needs to pay attention to:

1. "translate" is translating the cross-section, instead of the origin. So if you want to move the origin to the tension center (tc_2, tc_3), you need to "translate" the cross-section in the opposite direction
2. "translate" is done before "scale" ("s") and "rotate". This means that you need to un-scale the tension center coordinates to calculate the "translate" quantities.

In summary, the actual "translate" values for the input should be (-tc_2/s, -tc_3/s).

**Reply by vacupenda** [reply]
Thank you for your reply ,the problem has been solved!
